diff -Naur a/Makefile b/Makefile
--- a/Makefile	2021-04-09 11:48:40.000000000 -0300
+++ b/Makefile	2021-09-28 21:39:54.996031451 -0300
@@ -24,9 +24,9 @@
 CFLAGS_ALL := $(CFLAGS_AUTO) $(CFLAGS)
 LDFLAGS_ALL := $(LDFLAGS_AUTO) $(LDFLAGS)
 LDLIBS_ALL := $(LDLIBS_AUTO) $(LDLIBS)
-AR := $(CROSS_COMPILE)ar
-RANLIB := $(CROSS_COMPILE)ranlib
-STRIP := $(CROSS_COMPILE)strip
+AR := $(CROSS_COMPILE)$(AR_AUTO)
+RANLIB := $(CROSS_COMPILE)$(RANLIB_AUTO)
+STRIP := $(CROSS_COMPILE)$(STRIP_AUTO)
 INSTALL := ./tools/install.sh
 
 TYPES := size uid gid pid time dev ino
