diff -Naur a/RichString.c b/RichString.c
--- a/RichString.c	2021-01-11 14:18:21.000000000 -0300
+++ b/RichString.c	2021-04-10 02:59:33.909049996 -0300
@@ -57,7 +57,7 @@
    int newLen = from + len;
    RichString_setLen(this, newLen);
    for (int i = from, j = 0; i < newLen; i++, j++) {
-      this->chptr[i] = (CharType) { .attr = attrs & 0xffffff, .chars = { (iswprint(data[j]) ? data[j] : '?') } };
+      this->chptr[i] = (CharType) { .attributes = attrs & 0xffffff, .chars = { (iswprint(data[j]) ? data[j] : '?') } };
    }
 
    return wcswidth(data, len);
diff -Naur a/RichString.h b/RichString.h
--- a/RichString.h	2021-01-11 14:18:21.000000000 -0300
+++ b/RichString.h	2021-04-10 03:01:19.995044608 -0300
@@ -22,8 +22,8 @@
 #ifdef HAVE_LIBNCURSESW
 #define RichString_printVal(this, y, x) mvadd_wchstr(y, x, (this).chptr)
 #define RichString_printoffnVal(this, y, x, off, n) mvadd_wchnstr(y, x, (this).chptr + (off), n)
-#define RichString_getCharVal(this, i) ((this).chptr[i].chars[0] & 255)
-#define RichString_setChar(this, at, ch) do { (this)->chptr[(at)] = (CharType) { .chars = { ch, 0 } }; } while (0)
+#define RichString_getCharVal(this, i) ((this).chptr[i].vals[0] & 255)
+#define RichString_setChar(this, at, ch) do { (this)->chptr[(at)] = (CharType) { .vals = { ch, 0 } }; } while (0)
 #define CharType cchar_t
 #else
 #define RichString_printVal(this, y, x) mvaddchstr(y, x, (this).chptr)
