diff --git a/Makefile b/Makefile
index f742199..ec6a1e1 100644
--- a/Makefile
+++ b/Makefile
@@ -174,6 +174,7 @@ LIBCSRC=\
 	src/nix/readlink.c\
 	src/nix/rename.c\
 	src/nix/rmdir.c\
+	src/nix/rmpath.c\
 	src/nix/seek.c\
 	src/nix/stat.c\
 	src/nix/strtomode.c\
diff --git a/inc/tertium/fns.h b/inc/tertium/fns.h
index f63b56a..1ca8c7b 100644
--- a/inc/tertium/fns.h
+++ b/inc/tertium/fns.h
@@ -167,6 +167,7 @@ size c_nix_fdread(ctype_fd, void *, usize);
 size c_nix_readlink(char *, usize, char *);
 ctype_status c_nix_rename(char *, char *);
 ctype_status c_nix_rmdir(char *);
+ctype_status c_nix_rmpath(char *);
 ctype_status c_nix_seek(ctype_fd, ctype_fssize, int);
 ctype_status c_nix_stat(ctype_stat *, char *);
 uint c_nix_strtomode(char *, uint, uint);
diff --git a/src/nix/rmpath.c b/src/nix/rmpath.c
new file mode 100644
index 0000000..e4a8955
--- /dev/null
+++ b/src/nix/rmpath.c
@@ -0,0 +1,21 @@
+#include <tertium/cpu.h>
+#include <tertium/std.h>
+
+ctype_status
+c_nix_rmpath(char *dir)
+{
+	char *s;
+	char buf[C_PATHMAX];
+
+	c_str_cpy(buf, sizeof(buf), dir);
+	c_str_rtrim(buf, sizeof(buf), "/");
+
+	for (;;) {
+		if (c_nix_rmdir(buf) < 0)
+			return -1;
+		if (!(s = c_str_rchr(buf, sizeof(buf), '/')))
+			break;
+		*s = 0;
+	}
+	return 0;
+}
diff --git a/src/str/rtrim.c b/src/str/rtrim.c
index 6cfa260..fd78da1 100644
--- a/src/str/rtrim.c
+++ b/src/str/rtrim.c
@@ -6,12 +6,12 @@ c_str_rtrim(char *str, usize n, char *t)
 {
 	usize len, tlen;
 
-	len = c_str_len(str, n);
+	len = c_str_len(str, n) - 1;
 	tlen = c_str_len(t, -1);
 	for (; len; --len) {
 		if (!c_mem_chr(t, tlen, str[len]))
 			break;
-		str[len--] = 0;
+		str[len] = 0;
 	}
 	return str;
 }
