diff --git a/src/venus.c b/src/venus.c
index 686617c..feb181e 100644
--- a/src/venus.c
+++ b/src/venus.c
@@ -66,6 +66,7 @@ split_lines(char *s) {
 	c_arr_trunc(&arr, 0, sizeof(uchar));
 	if (c_dyn_cat(&arr, &s, 1, sizeof(char *)) < 0)
 		c_err_die(1, "c_dyn_cat");
+
 	for (; *s; ++s) {
 		if (*s == '\n') {
 			*s++ = 0;
@@ -73,6 +74,11 @@ split_lines(char *s) {
 				c_err_die(1, "c_dyn_cat");
 		}
 	}
+
+	s = nil;
+	if (c_dyn_cat(&arr, &s, 1, sizeof(char *)) < 0)
+		c_err_die(1, "c_dyn_cat");
+
 	return c_arr_data(&arr);
 }
 
@@ -442,7 +448,7 @@ pkgdel(char **argv)
 			/* TODO: deal with errors in a better way */
 			if ((c_nix_unlink(s) < 0) && errno != C_ENOENT)
 				c_err_die(1, "c_nix_unlink %s/%s", root, s);
-			c_nix_rmdir(c_gen_dirname(*args));
+			c_nix_rmpath(c_gen_dirname(s));
 		}
 		if ((c_nix_unlink(file) < 0) && errno != C_ENOENT)
 			c_err_die(1, "c_nix_unlink %s", file);
